indiserver.cpp

```diff
@@ -0,0 +1,3703 @@
+
+protected:
+    RemoteDvrInfo(const RemoteDvrInfo & model);
+
+public:
+    std::string host;
+    int port;
+
+    RemoteDvrInfo();
+    virtual ~RemoteDvrInfo();
+
+    virtual void start();
+
+    virtual RemoteDvrInfo * clone() const;
+
+    virtual const std::string remoteServerUid() const 
+    { 
+        return std::string(host) + ":" + std::to_string(port);
+    }
+};
+
+class TcpServer {
+    int port;
+    int sfd = -1;
+    ev::io sfdev;
+
+    /* prepare for new client arriving on socket.
+     * exit if trouble.
+     */
+    void accept();
+    void ioCb(ev::io &watcher, int revents);
+public:
+    TcpServer(int port);
+
+    /* create the public INDI Driver endpoint lsocket on port.
+     * return server socket else exit.
+     */
+    void listen();
+};
+
+class UnixServer {
+    std::string path;
+    int sfd = -1;
+    ev::io sfdev;
+
+    void accept();
+    void ioCb(ev::io & watcher, int revents);
+
+    virtual void log(const std::string & log) const;
+public:
+    UnixServer(const std::string & path);
+
+    /* create the public INDI Driver endpoint over UNIX (local) domain.
+     * exit on failure
+     */
+    void listen();
+};
+
+
+static void log(const std::string & log);
+/* Turn a printf format into std::string */
+static std::string fmt(const char * fmt, ...) __attribute__ ((format (printf, 1, 0)));
+
+static char *indi_tstamp(char *s);
+
+static const char *me;                                 /* our name */
+static int port = INDIPORT;                            /* public INDI port */
+static std::string unixSocketPath = INDIUNIXSOCK;
+static int verbose;                                    /* chattiness */
+static char *ldir;                                     /* where to log driver messages */
+static unsigned int maxqsiz  = (DEFMAXQSIZ * 1024 * 1024); /* kill if these bytes behind */
+static unsigned int maxstreamsiz  = (DEFMAXSSIZ * 1024 * 1024); /* drop blobs if these bytes behind while streaming*/
+static int maxrestarts   = DEFMAXRESTART;
+
+static std::vector<XMLEle *> findBlobElements(XMLEle * root);
+
+static void logStartup(int ac, char *av[]);
+static void usage(void);
+static void noSIGPIPE(void);
+static char *indi_tstamp(char *s);
+static void logDMsg(XMLEle *root, const char *dev);
+static void Bye(void);
+
+static int readFdError(int fd);                       /* Read a pending error condition on the given fd. Return errno value or 0 if none */
+
+static void * attachSharedBuffer(int fd, size_t & size);
+static void dettachSharedBuffer(int fd, void * ptr, size_t size);
+
+int main(int ac, char *av[])
+{
+    /* log startup */
+    logStartup(ac, av);
+
+    /* save our name */
+    me = av[0];
+
+#ifdef OSX_EMBEDED_MODE
+
+    char logname[128];
+    snprintf(logname, 128, LOGNAME, getlogin());
+    fprintf(stderr, "switching stderr to %s", logname);
+    freopen(logname, "w", stderr);
+
+    fifo = new Fifo();
+    fifo->name = FIFONAME;
+    verbose   = 1;
+    ac        = 0;
```

Comment:
Any particular reason why argument parsing is disabled in this mode?

Reply:
It is already the case on master branch...